src/orion/algo/base.py

```diff
@@ -319,12 +320,30 @@ def has_observed(self, point):
             True if the point's objective was observed by the algo, False otherwise.
 
         """
-
         trial_id = self.get_id(point)
         return (
             trial_id in self._trials_info and self._trials_info[trial_id][1] is not None
         )
 
+    @property
+    def unwrapped(self) -> "BaseAlgorithm":
+        return self
+
+    @contextmanager
+    def warm_start_mode(self):
+        """ Context manager that is used while using points from similar experiments to
+        bootstrap (warm-start) the algorithm.
+
+        The idea behing this is that we don't want the algorithm to modify its state the
+        same way it would if it were observing regular points. For example, the number
+        of "used" trials shouldn't increase, etc.
+
+        New Algorithms or Algo wrappers can implement this method to control how the
+        state of the algo is affected by observing trials from other tasks than the
+        current (target) task.
+        """
+        yield
```

Comment:
Should it raise NotImplementedError or running an algorithm with no implementation here could work? Could the default implementation be the temporary swapping of trials_info?

Reply:
Good question! It's better IMO to keep this a no-op as default. I think it's fine to assume that an algo can work with the trials from other tasks too.
The swapping of `trials_info` only happens for the Random algorithm, not for the other algorithms.